### ISSUE TITLE
Behavior changed during binding to collections in Net 8

### ISSUE THREAD

**Issue by bbenameur** [dotnet/runtime#97558]
### Description

Duplicated properties when deserialize array from appsetting.json only with .Net 8

### Configuration

on the appsetting.json i have an object with this structure:
```json
 "ArrayObject": [
   {  
     "Elements": 
       {
         "Type": "xxxx",
         "BaseUrl": "xxxxxxxx",
         "one": "",
         "xxxx": ""

       }
     
   }
 ]
```

There are a stranger behavior only with .Net 8
using
```csharp
var builder = WebApplication.CreateBuilder(args);

var result = builder.Configuration.GetSection("ArrayObject").Get<List<ArrayObject>>();
 the result this case will be 
```
### Regression?

![Net8](https://github.com/dotnet/core/assets/32399944/fae2a678-4e73-4b3e-b876-e0fa93a3d39c)

As you  see Elements should be null and empty but with Net 8 no it's take the number of properties on the elements objects.

### Other information

It's ok with Net 6 see image
![Net6](https://github.com/dotnet/core/assets/32399944/7b498a6b-9e5f-4c5f-871f-64dce757a208)

**Comment by bbenameur** [comment]
@ericstj

**Comment by bbenameur** [comment]
@eerhardt

**Comment by bbenameur** [comment]
@davidfowl

**Comment by eerhardt** [comment]
@bbenameur - can you post a complete repro? The code above doesn't compile since it doesn't define `ArrayObject`.

**Comment by bbenameur** [comment]
Hello @eerhardt
this is the project
https://github.com/bbenameur/demo-project
debug please and see the result on program.cs
var result = builder.Configuration.GetSection("ArrayObject").Get<List<ArrayObject>>();

**Comment by bbenameur** [comment]
Yes i confirm that the behavior is changed

**Comment by bbenameur** [comment]
@ericstj  Am I in the right group dotnet/runtime ?

**Comment by ericstj** [comment]
Yes this is the right group and area.  I'm updating the issue I understand more.  Thank you for the report.

This is the cause: https://github.com/dotnet/runtime/pull/91180/files#diff-2e6f75e1d4f577e1c8c1b6a32d158ce7812a38ad8fcd3b4d762157b92c03e890R426-R429

I read through that PR and didn't see _why_ we were adding another case here.  Moreover - we'll only create the instance and never bind any of it's properties which seems wrong.  If I change the type to have a constructor, then it will fail:
```C#
public class Element
{
	public Element(string type) { Type = type; }
	public string Type { get; set; }
}
```

```c#
var result = config.GetSection("ArrayObject").Get<List<ArrayObject>>(o => o.ErrorOnUnknownConfiguration = true);
```

```
System.InvalidOperationException: 'Cannot create instance of type 'Element' because parameter 'type' has no matching config. Each parameter in the constructor that does not have a default value must have a corresponding config entry.'
```

This is definitely a regression that we'll look into fixing.  @bbenameur - is this blocking you?  Seems like your JSON might have been set up in a way that didn't match your objects and you weren't noticing that until now.  Can you fix either the JSON or your type shape to make things work correctly?

**Comment by bbenameur** [comment]
Thanks @ericstj for your quick response, 
We were able to bypass this regression with a piece of code (not clean), but no blocking for the moment, 
please I would like to receive a notification when the problem is resolved

**Comment by tarekgh** [comment]
Looks we fixed this in 9.0 https://github.com/dotnet/runtime/pull/96737. We were watching to see if more reports on the issue before porting it to servicing. I'll confirm first the fix address this issue and then will port it to servicing.

**Comment by tarekgh** [comment]
Update:

https://github.com/dotnet/runtime/pull/96737 doesn't address this issue. I'll look more here at the right fix.

**Comment by ericstj** [comment]
I originally misidentified the regressing change, it was actually https://github.com/dotnet/runtime/pull/86485 which was fixing https://github.com/dotnet/runtime/issues/66229 - that makes more sense.

It still seems wrong to create an instance when operating on a config section with a value, that value cannot be directly assigned, and there are no child sections.  Maybe we only do this if the config.value is null as well.

**Comment by bbenameur** [comment]
The behavior is ok only if the element class has a constructor and without **o => o.ErrorOnUnknownConfiguration = true**
```csharp
public class Element
{
	public Element(string type) { Type = type; }
	public string Type { get; set; }
}
```

**Comment by tarekgh** [comment]
@bbenameur 

> The behavior is ok only if the element class has a constructor and without o => o.ErrorOnUnknownConfiguration = true

This is expected because the binder will not be able to create instance of the Element class. When `public Element(string type)` is not there, the binder can create the instance using the default constructor provided by the compiler.

**Comment by bbenameur** [comment]
Hello @tarekgh
So you confirm that it will be fixed on .Net 9?

**Comment by tarekgh** [comment]
@bbenameur For sure we are going to fix it in 9.0. We'll consider porting the fix back 8.0 if the changes are not risky and scoped. The fix for this is going to span the source generator too which I am currently looking at.

**Comment by tarekgh** [comment]
This is now fixed in 9.0. We'll watch if we'll get more reports and then consider porting the fix to 8.0.
